File: Core/Entities/Loan.cs
namespace Core.Entities
{
    public class Loan
    {
        public int LoanID { get; set; }
        public int ApplicationID { get; set; }
        public int CustomerID { get; set; }
        public int LoanProductID { get; set; }
        public decimal ApprovedAmount { get; set; }
        public DateTime DisbursementDate { get; set; }
        public DateTime MaturityDate { get; set; }
        public decimal InterestRate { get; set; }
        public decimal CurrentBalance { get; set; }
        public int OriginalTermMonths { get; set; }
        public string LoanStatus { get; set; }
        public DateTime? LastPaymentDate { get; set; }
        public DateTime? NextPaymentDueDate { get; set; }
    }
}

File: Angular/loan-management-system/src/app/models/loan.ts
import { LoanProduct } from './loan-product';

export interface Loan {
  loanID: number;
  applicationId: number;
  customerID: number;
  loanProductId: number;
  approvedAmount: number;
  disbursementDate: string;
  maturityDate: string;
  interestRate: number;
  currentBalance: number;
  originalTermMonths: number;
  loanStatus: string;
  lastPaymentDate: string | null;
  nextPaymentDueDate: string | null;
  loanProduct?: LoanProduct; // Optional for display purposes
}
